src/compo/viirs_aod2ioda.py

```diff
@@ -37,162 +40,335 @@
         self.varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
-        self._read()
+        self.setDicts()
+        self.read()
 
-    def _read(self):
+        DimDict['Location'] = len(self.outdata[('latitude', metaDataName)])
+        DimDict['Channel'] = np.array(channels)
+
+    def setDicts(self):
+        meta_keys = [m_item[0] for m_item in locationKeyList]
+        # Set units of the MetaData variables and all _FillValues.
+        self.varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+        for key in meta_keys:
+            dtypestr = locationKeyList[meta_keys.index(key)][1]
+            if locationKeyList[meta_keys.index(key)][2]:
+                self.varAttrs[(key, metaDataName)]['units'] = locationKeyList[meta_keys.index(key)][2]
+            self.varAttrs[(key, metaDataName)]['_FillValue'] = missing_vals[dtypestr]
+
+        var_keys = [v_item[0] for v_item in varsKeyList]
         # set up variable names for IODA
         for iodavar in obsvars:
-            self.varDict[iodavar]['valKey'] = iodavar, obsValName
-            self.varDict[iodavar]['errKey'] = iodavar, obsErrName
-            self.varDict[iodavar]['qcKey'] = iodavar, qcName
-            self.varAttrs[iodavar, obsValName]['coordinates'] = 'longitude latitude'
-            self.varAttrs[iodavar, obsErrName]['coordinates'] = 'longitude latitude'
-            self.varAttrs[iodavar, qcName]['coordinates'] = 'longitude latitude'
-            self.varAttrs[iodavar, obsValName]['_FillValue'] = -9999.
-            self.varAttrs[iodavar, obsErrName]['_FillValue'] = -9999.
-            self.varAttrs[iodavar, qcName]['_FillValue'] = -9999
-            self.varAttrs[iodavar, obsValName]['units'] = '1'
-            self.varAttrs[iodavar, obsErrName]['units'] = '1'
+            for key in var_keys:
+                varGroupName = varsKeyList[var_keys.index(key)][1]
+                dtypestr = varsKeyList[var_keys.index(key)][2]
+                coord = varsKeyList[var_keys.index(key)][3]
+                self.varDict[iodavar][key] = iodavar, varGroupName
+                self.varAttrs[iodavar, varGroupName]['coordinates'] = coord
+                self.varAttrs[iodavar, varGroupName]['_FillValue'] = missing_vals[dtypestr]
+                if varsKeyList[var_keys.index(key)][4]:
+                    self.varAttrs[iodavar, varGroupName]['units'] = varsKeyList[var_keys.index(key)][4]
+
+    def get_platform_sensor_names(self):
+        if self.provider == 'noaa':
+            satellite = self.glb_attrs["satellite_name"]
+            sensor = self.glb_attrs["instrument_name"]
+        elif self.provider == 'nasa':
+            satellite = self.glb_attrs["platform"]
+            sensor = self.glb_attrs["instrument"]
+
+        if 'NPP' in satellite:
+            AttrData["platform"] = "suomi_npp"
+            AttrData["sensor"] = "v.viirs-m_npp"
+        elif satellite == 'NOAA-20':
```

Comment:
N21?

Reply:
Thank you, @CoryMartin-NOAA, I just added it. 
NASA has not released the VIIRS AOD product of NOAA 21, so I did not add it.